runtime/src/committee.rs

```diff
@@ -0,0 +1,752 @@
+        type Origin = Origin;
+        type Trait = Test;
+        type Error = DispatchError;
+
+        fn dispatch(self, _origin: Self::Origin) -> DispatchResult<Self::Error> {
+            Ok(())
+        }
+    }
+
+    impl identity::Trait for Test {
+        type Event = ();
+        type Proposal = TestProposal;
+        type AcceptTickerTransferTarget = Test;
+    }
+
+    impl crate::asset::AcceptTickerTransfer for Test {
+        fn accept_ticker_transfer(_: IdentityId, _: u64) -> srml_support::dispatch::Result {
+            unimplemented!()
+        }
+    }
+
+    type Identity = identity::Module<Test>;
+    type AccountId = <AnySignature as Verify>::Signer;
+
+    parameter_types! {
+        pub const CommitteeOrigin: AccountId = AccountId::from(AccountKeyring::Alice);
+    }
+
+    impl Trait<Instance1> for Test {
+        type Origin = Origin;
+        type Proposal = Call;
+        type CommitteeOrigin = EnsureSignedBy<CommitteeOrigin, AccountId>;
+        type Event = ();
+    }
+
+    impl Trait for Test {
+        type Origin = Origin;
+        type Proposal = Call;
+        type CommitteeOrigin = EnsureSignedBy<CommitteeOrigin, AccountId>;
+        type Event = ();
+    }
+
+    pub type Block = sr_primitives::generic::Block<Header, UncheckedExtrinsic>;
+    pub type UncheckedExtrinsic = sr_primitives::generic::UncheckedExtrinsic<u32, u64, Call, ()>;
+
+    srml_support::construct_runtime!(
+		pub enum Test where
+			Block = Block,
+			NodeBlock = Block,
+			UncheckedExtrinsic = UncheckedExtrinsic
+		{
+			System: system::{Module, Call, Event},
+			Committee: committee::<Instance1>::{Module, Call, Event<T>, Origin<T>, Config<T>},
+			DefaultCommittee: committee::{Module, Call, Event<T>, Origin<T>, Config<T>},
+		}
+	);
+
+    fn make_ext() -> sr_io::TestExternalities<Blake2Hasher> {
+        GenesisConfig {
+            committee_Instance1: Some(committee::GenesisConfig {
+                members: vec![
+                    IdentityId::from(1),
+                    IdentityId::from(2),
+                    IdentityId::from(3),
+                ],
+                vote_threshold: (ProportionMatch::AtLeast, 1, 1),
+                phantom: Default::default(),
+            }),
+            committee: None,
+        }
+        .build_storage()
+        .unwrap()
+        .into()
+    }
+
+    #[test]
+    fn motions_basic_environment_works() {
+        with_externalities(&mut make_ext(), || {
+            System::set_block_number(1);
+            assert_eq!(
+                Committee::members(),
+                vec![
+                    IdentityId::from(1),
+                    IdentityId::from(2),
+                    IdentityId::from(3)
+                ]
+            );
+            assert_eq!(Committee::proposals(), Vec::<H256>::new());
+        });
+    }
+
+    fn make_proposal(value: u64) -> Call {
+        Call::System(system::Call::remark(value.encode()))
+    }
+
+    fn make_account(
```

Comment:
There is something similar in `test\storage.rs`.

Reply:
Because of how DefaultInstance is setup, the return type of `make_account` is the Origin from `Test` which is setup using `construct_runtime`. Plus for this test, we don't need balances. That's the reason I've made a function to just register DIDs.